### ISSUE TITLE
Texture/Sprite Filtering

### ISSUE THREAD

**Issue by Papermanzero** [Arisotura/melonDS#986]
It would be great to see texture filtering and sprite filtering for 3D based games. bilinear and bicubic filtering would increase the presentation of the games on higher resolutions.

**Comment by poudink** [comment]
some work for this was done on the blackmagic_II branch but it's so old I think it might be scrapped

**Comment by zephryn** [comment]
I've been poking at the OpenGL 3D renderer code a bit and the shaders already include a linear filtering option for 3D textures that's (as far as I can tell) never used but seems to work alright. Almost got it working with the frontend by adding a checkbox in the "Video settings" dialog and passing that to a new field in the shader UBO, but I must've been a bit naive with how I went about it since it never really ended up passing the parameter properly.

**Comment by Papermanzero** [comment]
It would be awesome to get texture filtering with opengl and vulkan. Bilinear, Bicubic, xbrz, etc

**Comment by sandwichwater** [comment]
pretty sure the compute shader renderer will support texture replacement

**Comment by Papermanzero** [comment]
> pretty sure the compute shader renderer will support texture replacement

Hopefully also filtering. Zelda with smooth textures and high res would look awesome

**Comment by sandwichwater** [comment]
hmm ive tried zelda with texture replacement and no filtering and it looked perfect to me , though different people do have different preferences so maybe it would be nice

**Comment by Papermanzero** [comment]
Thats true. I would like to keep the original look but just have the pixels of the textures smooth
